services/context-box/server/cbox.go

```diff
@@ -66,6 +67,36 @@ func (ci *ConfigInfo) HasError() bool {
 	return false
 }
 
+// hasDestroyError returns true if any cluster errored while destroying.
+func (ci *ConfigInfo) hasDestroyError() bool {
+	for _, v := range ci.State {
+		if v.Status == pb.Workflow_ERROR.String() && strings.Contains(v.Stage, "DESTROY") {
+			return true
+		}
+	}
+	return false
+}
+
+// scheduleDeletion returns true if config should be pushed onto any queue due to deletion.
+// This ignores the build errors, as we want to remove infrastructure if secret was deleted,
+// however, respects error from destroy workflow, as we do not want to retry indefinitely.
+func (ci *ConfigInfo) scheduleDeletion() bool {
+	if ci.hasDestroyError() {
+		// Ignore as deletion already errored out
+		return false
+	}
+	if ci.MsChecksum == nil && ci.DsChecksum != nil {
+		// Scheduler queue
+		return true
+	}
+	if ci.MsChecksum == nil && ci.DsChecksum == nil && ci.CsChecksum != nil {
+		// Builder queue
+		return true
+	}
+	// Not scheduled for deletion
+	return false
+}
```

Comment:
Do I understand the deletion correctly? Scheduler will enqueue configs scheduled for deletion and builder will try to execute the destroy of the config 3x before failure.

Reply:
Yes, the current workflow is 

secret gets deleted -> Frontend sets msChecksum to nil, -> Context-box pushes config to SchedulerQueue ->  Scheduler sets dsChecksum to nil -> Context-box pushes config to BuilderQueue -> Builder will try to remove it 3 times before aborting with an error if any.